Analytics/Classes/Internal/SEGSegmentIntegration.m

```diff
@@ -461,9 +461,8 @@ - (void)enqueueAction:(NSString *)action dictionary:(NSMutableDictionary *)paylo
 - (void)queuePayload:(NSDictionary *)payload
 {
     @try {
-        // We only queue upto 1000 items, so trim the queue to 1000-1=999
-        // before we add a new element.
-        trimQueue(self.queue, 999);
+        // Trim the queue to maxQueueSize - 1 before we add a new element.
+        trimQueue(self.queue, self.analytics.configuration.maxQueueSize - 1);
```

Comment:
what if somebody sets `maxQueueSize` to 0?

Reply:
Oh, good catch! I guess we have to choose from one of these behaviours (ordered with what I think is best first) : 
1. forbid using `maxQueueSize = 0` and throw an error in `maxQueueSize` setter if 0 is set
2. when `maxQueueSize == 0` ignore all events
3. when `maxQueueSize == 0` behave like `maxQueueSize == 1`

Reply:
Option 4: `maxQueueSize == 0` should behave like a no-op, i.e. use the default setting.

Reply:
Option 5: Document that this must be > 1, and when `maxQueueSize == 0` the behaviour is unspecified. I think this is reasonable for now.